client/config/config.go

```diff
@@ -115,35 +117,42 @@ type NonvotingAuthority struct {
 	// Address is the IP address/port combination of the authority.
 	Address string
 
-	// PublicKey is the authority's public key.
-	PublicKey *eddsa.PublicKey
+	// IdentityPublicKeyPem is the authority's identity public key pem.
+	IdentityPublicKeyPem string
 
-	// LinkPublicKeyPem is the absolute file path to the
-	// authority's link public key.
+	// LinkPublicKeyPem is the PEM data of the authority's link public key.
 	LinkPublicKeyPem string
 }
 
 // New constructs a pki.Client with the specified non-voting authority config.
 func (nvACfg *NonvotingAuthority) New(l *log.Backend, pCfg *proxy.Config, linkKey wire.PrivateKey, datadir string) (pki.Client, error) {
-	scheme := wire.NewScheme()
-	authLinkKey, err := scheme.PublicKeyFromPemFile(filepath.Join(datadir, nvACfg.LinkPublicKeyPem))
+	scheme := wire.DefaultScheme
+
+	authLinkPrivKey := scheme.GenerateKeypair(rand.Reader)
+	authLinkKey := authLinkPrivKey.PublicKey()
+	err := pem.FromPEMString(nvACfg.LinkPublicKeyPem, authLinkKey)
 	if err != nil {
 		return nil, err
 	}
+
+	_, identityPublicKey := cert.Scheme.NewKeypair()
+	pem.FromPEMString(nvACfg.IdentityPublicKeyPem, identityPublicKey)
+
+	linkHash := blake2b.Sum256(linkKey.PublicKey().Bytes())
 	cfg := &nvClient.Config{
-		AuthorityLinkKey: authLinkKey,
-		LinkKey:          linkKey,
-		LogBackend:       l,
-		Address:          nvACfg.Address,
-		PublicKey:        nvACfg.PublicKey,
-		DialContextFn:    pCfg.ToDialContext("nonvoting:" + nvACfg.PublicKey.String()),
+		AuthorityLinkKey:     authLinkKey,
+		LinkKey:              linkKey,
+		LogBackend:           l,
+		Address:              nvACfg.Address,
+		AuthorityIdentityKey: identityPublicKey,
+		DialContextFn:        pCfg.ToDialContext(fmt.Sprintf("nonvoting: %x", linkHash[:])),
```

Comment:
I wonder if we should consider just putting some random data here instead. Like do we care about telling Tor what our linkKey or not? If we just send some random stuff then we never have to answer that question. Thoughts?

Reply:
let's do it in another ticket/pr

Reply:
opened https://github.com/katzenpost/katzenpost/issues/56